flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/GroupReduceITCase.java

```diff
@@ -434,6 +431,95 @@ public void testCorrectnessOfGroupreduceWithDescendingGroupSort() throws Excepti
 	}
 
 	@Test
+	public void testCorrectnessOfGroupreduceWithExplicitPartitionOfReducer() throws Exception {
+		/*
+		 * check correctness of groupReduce with descending group sort
+		 */
+		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		env.setParallelism(1);
```

Comment:
Is it ok if I modify the existing test testCorrectnessOfGroupReduceOnTuplesWithCombine() and add a partitionByHash to the input dataset? I tested and debugged it and it seems to work fine after the changes and gives the same expected result as given by testCorrectnessOfGroupReduceOnTuplesWithCombine().